src/npf/rproc/npf_ext_dpi.c

```diff
@@ -178,9 +178,15 @@ dpi_match(npf_cache_t *npc, struct rte_mbuf *mbuf, const struct ifnet *ifp,
 	/* Find or attach the DPI flow info. Do first packet inspection */
 	struct dpi_flow *dpi_flow = npf_session_get_dpi(se);
 	if (!dpi_flow) {
+#ifdef USE_NDPI
 		uint8_t engines[] = {IANA_USER, IANA_NDPI};
+		size_t engines_len = 2;
+#else
+		uint8_t engines[] = {IANA_USER};
+		size_t engines_len = 1;
```

Comment:
@dfawcus

ARRAY_SIZE, as earlier.

Reply:
I think this repeated code should be in a little helper function so USE_NDPI is kept to a minimum, and there's only one place to add additional DPI engines in future.